src/features/upload-file/providers/gcp-provider.ts

```diff
@@ -0,0 +1,78 @@
+import { UploadedFile } from 'admin-bro'
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { UploadResponse, DeleteFileResponse } from '@google-cloud/storage'
+import { BaseProvider } from './base-provider'
+
+/**
+ * Google Storage options which can be set for GCP file upload.
+ * In order to setup GCP credentials you have to follow this
+ * {@link https://cloud.google.com/docs/authentication/getting-started tutorial}.
+ * Basically it comes down to downloading service account and setting GOOGLE_APPLICATION_CREDENTIALS
+ * env variable. After that you are ready to go.
+ * @memberof module:@admin-bro/upload
+ */
+export type GCPOptions = {
+  /**
+   * Google Storage Bucket name, where files will be stored
+   */
+  bucket: string
+  /**
+   * indicates how long links should be available after page load (in minutes).
+   * Default to 24h. If set to 0 adapter will mark uploaded files as public.
+   */
+  expires?: number;
+}
+
+export class GCPProvider extends BaseProvider {
+  private storage: Storage
+
+  public expires: number
+
+  constructor(options: GCPOptions) {
+    super(options.bucket)
+
+    let GCPStorage: typeof Storage
+    try {
+      // eslint-disable-next-line
+      GCPStorage = require('@google-cloud/storage').Storage
```

Comment:
Optional: doesn't matter in this case but this should work too and will eliminate eslint warning
```suggestion
      import('@google-cloud/storage').then(({Storage}) => { this.GCPStorage = new Storage() }).catch([...])
```